kroxylicious/src/main/java/io/kroxylicious/proxy/internal/InternalCompletableFuture.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+/**
+ * Implementation of CompletableFuture that uses the given {@link Executor}
+ * for async method invocations that do not specify one.
+ * <br/>
+ * @param <T> The result type returned by this future's {@code join}
+ * and {@code get} methods.
+ * @see InternalCompletionStage
+ */
+class InternalCompletableFuture<T> extends CompletableFuture<T> {
+
+    private final Executor executor;
+
+    InternalCompletableFuture(Executor executor) {
+        Objects.requireNonNull(executor);
+        this.executor = executor;
```

Comment:
nit these can be conbined.
```suggestion
        this.executor = Objects.requireNonNull(executor);
```